database/backend/design-patterns.json

```
@@ -0,0 +1,156 @@
+[
+    {
+        "name": "Factory Method",
+        "description": "Factory Method is a creational design pattern that provides an interface for creating objects in a superclass, but allows subclasses to alter the type of objects that will be created.",
+        "url": "https://refactoring.guru/design-patterns/factory-method",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Abstract Factory",
+        "description": "Abstract Factory is a creational design pattern that lets you produce families of related objects without specifying their concrete classes.",
+        "url": "https://refactoring.guru/design-patterns/abstract-factory",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Builder",
+        "description": "Builder is a creational design pattern that lets you construct complex objects step by step. The pattern allows you to produce different types and representations of an object using the same construction code.",
+        "url": "https://refactoring.guru/design-patterns/builder",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Prototype",
+        "description": "Prototype is a creational design pattern that lets you copy existing objects without making your code dependent on their classes.",
+        "url": "https://refactoring.guru/design-patterns/prototype",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Singleton",
+        "description": "Singleton is a creational design pattern that lets you ensure that a class has only one instance, while providing a global access point to this instance.",
+        "url": "https://refactoring.guru/design-patterns/singleton",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Adapter",
+        "description": "Adapter is a structural design pattern that allows objects with incompatible interfaces to collaborate.",
```

[Nitpick] Hmmm..."collaborate" makes this description a bit unclear. Does this pattern let object with different interfaces connect to each other?
```suggestion
        "description": "Adapter is a structural design pattern that allows objects with incompatible interfaces to collaborate.",
```

---

FIxed changes